### PR TITLE
Login + Signup feature

### PR DESCRIPTION
**Pull request by bryanloh-jw**
Creating pull request for Login and Signup feature

Frontend:
- Initialized with landing page
- Form for login + signup
- Input validation + error handling

Backend:
- Spring framework
- Authentication filter
- Endpoints for login + signup

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -1,10 +1,27 @@
 import { RouterProvider, createBrowserRouter } from 'react-router-dom';
+import { QueryClientProvider, QueryClient } from '@tanstack/react-query';
 import { Landing } from './pages';
+import { Overlay } from './components';
+import { Provider } from 'react-redux';
+import { store } from './store';
```
Comment (bryanloh-jw) [inline client/src/App.tsx:6]: nit: sort imports by dependencies & internal

File: client/src/api/axios.ts
Comment (bryanloh-jw) [inline client/src/api/axios.ts:1]: remove since not using

File: client/src/components/login/LoginForm.tsx [hunk 2]
```diff
@@ -0,0 +1,193 @@
+import { useForm } from 'react-hook-form';
```
Comment (Bryan-LohJW) [inline client/src/components/login/LoginForm.tsx:1]: nit: sort imports

File: client/src/components/login/LoginForm.tsx [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+import { z } from 'zod';
+import { Card } from './../ui/Card';
+import logo from '/src/assets/logo-light.png';
+import classes from './LoginForm.module.scss';
+import { useState } from 'react';
+import { useMutation } from '@tanstack/react-query';
+import { useOutsideClick } from '../../hooks';
+import { useDispatch } from 'react-redux';
+import { hideLogin, showSignUp, login as storeLogin } from '../../store';
+import { BiError } from 'react-icons/bi';
```
Comment (bryanloh-jw) [inline client/src/components/login/LoginForm.tsx:12]: nit: sort imports

File: client/src/components/signup/SignupForm.tsx [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+import classes from './SignUpForm.module.scss';
+import { useState } from 'react';
+import { useOutsideClick } from '../../hooks';
+import { useDispatch } from 'react-redux';
+import { hideSignUp, showLogin } from '../../store';
+import { BiError } from 'react-icons/bi';
+
+const schema = z.object({
+	email: z.string().email('Invalid email'),
+	password: z.string().min(8, 'Password should be at least 8 characters'),
```
Comment (bryanloh-jw) [inline client/src/components/signup/SignupForm.tsx:15]: change to follow login form

File: client/src/components/login/LoginForm.tsx [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+const schema = z.object({
+	email: z.string().email('Invalid email'),
+	password: z.string().min(8, 'Password is too short'),
+});
+
+type LoginData = {
```
Comment (bryanloh-jw) [inline client/src/components/login/LoginForm.tsx:19]: rename 'LoginData' to 'LoginInput' for clarity

File: client/src/components/login/LoginForm.tsx [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+
+	const {
+		register,
+		handleSubmit,
+		watch,
+		formState: { errors },
+		clearErrors,
+	} = useForm<LoginData>({ resolver: zodResolver(schema) });
+
+	const login = async (credentials: { email: string; password: string }) => {
```
Comment (bryanloh-jw) [inline client/src/components/login/LoginForm.tsx:41]: use the LoginInput type

File: client/src/components/login/LoginForm.tsx [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+			}
+		);
+		if (!response.ok) {
+			if (response.status === 403) {
+				setFormError('Invalid credentials');
+				return;
+			}
+			setFormError('Unexpected error, try again later');
+			return;
+		}
```
Comment (bryanloh-jw) [inline client/src/components/login/LoginForm.tsx:62]: if (response.status === 403) {}
if (!response.ok) {}

unnecessary nesting of logic that makes code unreadable

File: client/src/components/signup/SignupForm.tsx [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+});
+
+type SignUpData = {
+	email: string;
+	password: string;
+};
```
Comment (bryanloh-jw) [inline client/src/components/signup/SignupForm.tsx:21]: sign up input for clarity

File: client/src/components/signup/SignupForm.tsx [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+	register('email', {
+		onChange: () => {
+			setFormError(null);
+		},
+		onBlur: handleEmailBlur,
+	});
+
+	register('password', {
+		onBlur: handlePasswordBlur,
+	});
```
Comment (bryanloh-jw) [inline client/src/components/signup/SignupForm.tsx:100]: implement for login form as well, and remove onchange set form error null
when input change the validation will kick in again, so no point to remove error when user will see error again before submitting

File: client/src/components/ui/Card/Card.tsx [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+type CardProps = {
+	height?: string;
+	width?: string;
+};
+
+export const Card: FC<PropsWithChildren<CardProps>> = ({
+	children,
+	height,
+	width,
+}) => {
```
Comment (bryanloh-jw) [inline client/src/components/ui/Card/Card.tsx:13]: remove props, card will only be a wrapper and sizing will be handled by children

File: client/src/components/ui/Footer/Footer.tsx [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+		return new Date().getFullYear().toString();
+	};
+	return (
+		<footer className={classes.footer}>
+			<p>Ⓒ Copyright {`${year()}`}</p>
+			<BsGithub
+				className={classes.icon}
+				onClick={() => {
+					window.open(import.meta.env.VITE_GITHUB_LINK);
+					console.log('hello');
```
Comment (bryanloh-jw) [inline client/src/components/ui/Footer/Footer.tsx:15]: remove console.log

File: client/src/components/ui/Overlay/Overlay.module.scss
Comment (bryanloh-jw) [inline client/src/components/ui/Overlay/Overlay.module.scss:1]: remove

File: client/src/store/authStore.ts [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+	name: 'auth',
+	initialState,
+	reducers: {
+		login: (state, action: PayloadAction<LoginParams>) => {
+			if (!action.payload.token) return;
+			state.token = action.payload.token;
+			const payload = jose.decodeJwt(action.payload.token);
+			state.sub = payload.sub || null;
+			state.authenticated = true;
+			console.log('hello');
```
Comment (bryanloh-jw) [inline client/src/store/authStore.ts:31]: remove console log

File: database/01-init-db.sql
Comment (bryanloh-jw) [inline database/01-init-db.sql:1]: delete as not using

File: database/Dockerfile
Comment (bryanloh-jw) [inline database/Dockerfile:1]: delete as not using

File: server/create.sql
Comment (bryanloh-jw) [inline server/create.sql:1]: delete

File: server/docker-compose.yml [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+      - bryanloh
+  pgadmin:
+    image: dpage/pgadmin4
+    ports:
+      - 8000:80
+    environment:
+      PGADMIN_DEFAULT_EMAIL: postgres@hotmail.com
+      PGADMIN_DEFAULT_PASSWORD: password
+    networks:
+      - bryanloh
```
Comment (bryanloh-jw) [inline server/docker-compose.yml:21]: can comment out pg admin

File: server/src/main/java/com/bryan/commuteconnect/config/JwtService.java [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+import java.security.Key;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+@Service
+public class JwtService {
+
+    private static final String SECRET_KEY = "614E645266556A586E3272357538782F413F4428472B4B6250655368566B5970";
```
Comment (bryanloh-jw) [inline server/src/main/java/com/bryan/commuteconnect/config/JwtService.java:20]: put the authentication key in env file

File: server/src/main/java/com/bryan/commuteconnect/config/SecurityConfiguration.java [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+@EnableWebSecurity
+@RequiredArgsConstructor
+public class SecurityConfiguration {
+    private final JwtAuthenticationFilter jwtAuthFilter;
+    private final AuthenticationProvider authenticationProvider;
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        http
+//                .cors(Customizer.withDefaults())
```
Comment (bryanloh-jw) [inline server/src/main/java/com/bryan/commuteconnect/config/SecurityConfiguration.java:30]: remove comment

File: server/src/main/java/com/bryan/commuteconnect/config/SecurityConfiguration.java [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+                .cors(cors -> cors.disable())
+                .csrf().disable()
+                .authorizeHttpRequests()
+                .requestMatchers("/auth/**", "/demo")
```
Comment (bryanloh-jw) [inline server/src/main/java/com/bryan/commuteconnect/config/SecurityConfiguration.java:34]: remove demo

File: server/src/main/java/com/bryan/commuteconnect/demo/DemoController.java
Comment (bryanloh-jw) [inline server/src/main/java/com/bryan/commuteconnect/demo/DemoController.java:1]: remove demo

**Pull request merged**